Rubberduck.Parsing/ComReflection/ComModule.cs

```diff
@@ -59,7 +58,9 @@ private void GetComFields(ITypeInfo info, TYPEATTR attrib)
                     info.GetNames(desc.memid, names, names.Length, out int length);
                     Debug.Assert(length == 1);
 
-                    _fields.Add(new ComField(this, info, names[0], desc, index, DeclarationType.Constant));
+                    var type = desc.desc.lpvarValue == IntPtr.Zero ? DeclarationType.Variable : DeclarationType.Constant;
```

Comment:
This feels wrong - The second `desc` returns [an union](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.comtypes.vardesc.descunion?view=netframework-4.7.2). We need to ensure that it is appropriate to read from the `lpvarvalue` or we will get a bogus value. 

Based on [this documentation](https://msdn.microsoft.com/en-us/library/cc237860.aspx?f=255&MSPPError=-2147217396), we should be first reading from `VARKIND` and accessing the `lpvarValue` only if it is `VAR_CONST`. Note that if it's `VAR_PERINSTANCE`, it may be non-zero for the `oInst`, which is shared with the `lpvarValue`.

Reply:
That is basically what I wrote in chat earlier today. (I did not fully go through the documentation when hacking this together; I just did what made sense from the problem I encountered and the fact that the documentation states that `lpvarValue` _describes a constant_.)

I have to test whether the check for the `VARKIND` breaks resolving the actual references.

Reply:
Using the `VARKIND` is no option because VBA constants are exposed with `VAR_STATIC`.